.github/workflows/check_code_style.yml

```diff
@@ -0,0 +1,24 @@
+name: Check Code Style
+
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    branches:
+      - "main" 
```

Comment:
I've always been curious about this, why do we need both `push` and `pull_request`? Is the idea that if you push to a branch (new or otherwise) and then create a pull request from that branch the workflow that ran when you did the `push` won't be associated with the pull request? Can you explain to me why we need both? Because I think we only need the `push`.